build_runner/lib/src/server/hot_reload_client.dart

```diff
@@ -2,14 +2,85 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:async';
+import 'dart:convert';
 import 'dart:html';
+import 'dart:js';
 
+final _assetsDigestPath = r'$assetDigests';
 final _buildUpdatesProtocol = r'$livereload';
 
-main() {
+JsObject _dartLoader = context['\$dartLoader'];
+JsObject _urlToModuleId = _dartLoader['urlToModuleId'];
+
+List<String> get moduleUrls {
+  JsArray callMethod = context['Array']
+      .callMethod('from', [_urlToModuleId.callMethod('keys', [])]);
+  return List.castFrom(callMethod);
+}
+
+String moduleIdByPath(String path) =>
+    _urlToModuleId.callMethod('get', [window.location.origin + '/' + path])
+        as String;
+
+Future<JsObject> reloadModule(String moduleId) {
+  var completer = Completer<JsObject>();
+  _dartLoader.callMethod('forceLoadModule', [
+    moduleId,
+    JsFunction.withThis((_this, JsObject module) => completer.complete(module))
+  ]);
+  return completer.future;
+}
+
+class ReloadHandler {
+  final String Function(String) _moduleIdByPath;
+  final FutureOr<JsObject> Function(String) _reloadModule;
+  final Map<String, String> _digests;
+
+  ReloadHandler(Map<String, String> this._digests,
```

Comment:
you don't need to add the explicit type here, it will be the type of the field already (didn't even realize you could! haha)